sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java

```diff
@@ -123,9 +123,17 @@ public static Resource getExpectedRootVolume(
     }
 
     public static Resource getDesiredResource(String role, String principal, String name, Value value) {
+        return getDesiredResource(role, principal, name, value, "");
+    }
+
+    public static Resource getDesiredResource(String role,
```

Comment:
Should the naming here change? I figure `Desired` necessarily implies that `resourceId == ""`, otherwise it's expected, since we know the `resourceId` we want.